### PR TITLE
Fix skybox global color shader

### PR DESCRIPTION
**Pull request by zzmp**
- Removed vestigial ambientSphere preset code from DeferredLightingEffect
- Track current skyboxTexture mipmapLevels through to the gpu from DeferredLightingEffect
- Fix evalSkyboxGlobalColor shader proc to actually reflect the skybox, using mipmaps to make less glossy things reflect the blending of multiple skybox pixels

Functionally, this fixes lighting such that things with gloss are actually glossy, and reflect the skybox texture surrounding them. This is wired in as the ambient light, so the effect is modulated by the zone's ambient light intensity.

If you have a something reflective on a floor, it may reflect the bottom of the skybox, which looks borked. The skyboxGlobalColor will respect alpha values in the skybox, so setting an ambientTexture with alpha values on the bottom half, or a neutral, floor-like color on the bottom half, may get around this.

### CODE REVIEW

**Comment by zzmp** [comment]
#### Testing

Go to a place with a zone. Add some objects with roughness maps. Best testing would be with a plane with gradiated roughness so you can see the changes in reflection clarity. Try changing these things, and make sure they behave:
- Change the zone's skybox texture: it should reflect that texture.
- Change the zone's ambientURL texture: it should reflect that texture.
- Remove the zone's ambientURL texture: it should go back to skybox.
- Remove the zone's skybox: it should act as if it just had ambient light - no reflection.
- Add some texture back.
- Change the ambient light intensity - should change the intensity of the reflection.
- Change the light (not ambient) - it should show up with the reflection.

File: libraries/render-utils/src/DeferredGlobalLight.slh [hunk 1]
```diff
@@ -14,106 +14,83 @@
...
 
-// Everything about light
-<@include model/Light.slh@>
-
-<@func declareEvalAmbientGlobalColor()@>
-vec3 evalAmbientGlobalColor(mat4 invViewMat, float shadowAttenuation, float obscurance, vec3 position, vec3 normal, vec3 diffuse, vec3 specular, float gloss) {
-
-    // Need the light now
-    Light light = getLight();
+<@func prepareGlobalLight()@>
```
Comment (zzmp) [inline libraries/render-utils/src/DeferredGlobalLight.slh]: Is this up to our coding standard? It's basically a #DEFINE in scribe.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4eb975a90dd6f2482a72bb6eae22f9e305fe9082 available at: https://gist.github.com/6763b4d502c06330d4ec<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7201.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7201.dmg

**Comment by OmegaHeron** [comment]
Don't take this as a fail, because it **is not** but some tricks done to make certain things look certain ways will need re-thinking with this one.  ;)  Via trying to fake something under old conditions I seem to have made near a perfect test for this PR working.

This PR

hifi-snap-by-omegaheron-on-2016-02-25_17-54-56

Current

hifi-snap-by-omegaheron-on-2016-02-25_17-55-44

**Comment by OmegaHeron** [comment]
It would be good for others with a better overall understanding of how this works (should work etc) to test.  It's really not my line of expertise, but, I'm learning, rapidly.

**Comment by imgntn** [comment]
@OmegaHeron dang, been working out?!

**Comment by OmegaHeron** [comment]
lolol!  one of these days I have to tune weight paints a touch from defaults Mixamo generated.... with standing pose here it kinda gets lumpy about the arms.

**Comment by samcake** [comment]
THis has been merged in one of my branch so we merge to hifi/master in a common pr

**Pull request closed without merging**